packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { AzureClient } from "@fluidframework/azure-client";
+import { ConnectionState } from "@fluidframework/container-loader";
+import { ContainerSchema } from "@fluidframework/fluid-static";
+import { SharedTree, TreeConfiguration, SchemaFactory } from "@fluidframework/tree";
+import { timeoutPromise } from "@fluidframework/test-utils/internal";
+
+import { createAzureClient } from "./AzureClientFactory.js";
+
+const sf = new SchemaFactory("d302b84c-75f6-4ecd-9663-524f467013e3");
+
+/**
+ * Define a class that is an array of strings
+ * This class is used to create an array in the SharedTree
+ */
+class StringArray extends sf.array("StringArray", sf.string) {
+	/**
+	 * Remove the first item in the list if the list is not empty
+	 */
+	public removeFirst(): void {
+		if (this.length > 0) this.removeAt(0);
+	}
+
+	/**
+	 * Add an item to the beginning of the list
+	 */
+	public insertNew(str: string): void {
+		this.insertAtStart(str);
+	}
+}
+
+/**
+ * This object is passed into the SharedTree via the schematize method.
+ */
+const treeConfiguration = new TreeConfiguration(
+	// Specify the root type - StringArray.
+	StringArray,
```

Comment:
Any other types we could consider here? Probably variations of tests could focus on edits/reads of various types.

Reply:
https://fluidframework.com/docs/data-structures/tree/#node-types